### PR TITLE
Add ability to add scanner from new scan page if not configured yet.

### PR DESCRIPTION
**Pull request by elyousfi5**
[Screencast from 2024-07-29 09-58-51.webm](https://github.com/user-attachments/assets/ac3f8499-be73-4c6c-94ba-57482ae201b8)

### CODE REVIEW

**Review by nmasdoufi-ol — APPROVED** [review]

**Review by rabsondev — COMMENTED** [review]
Not sure about having a **New Scanner** button there, looks out of place 🤔

**Review by adnaneserrar — APPROVED** [review]

File: project/scanners/components/ScannerSelect.vue [hunk 1]
```diff
@@ -1,53 +1,110 @@
...
+        >
+          {{ item.raw.name }} (<code>{{ item.raw.endpoint }}</code>)
+        </p>
+        <code
+          v-else
+          class="mb-0"
+        >{{ item.raw.endpoint }}</code>
+      </template>
+    </v-select>
+    <div v-else>
```
Comment (mohsinenar) [inline project/scanners/components/ScannerSelect.vue:31]: I agree with rabson that does not seem in the right pleace!

File: project/scanners/components/ScannerSelect.vue [hunk 2]
```diff
@@ -1,53 +1,110 @@
...
-    return-object
-    label="Select a scanner"
-    placeholder="Select or create a scanner to run the scan on"
-    prepend-icon="mdi-shield-search"
-  >
-    <template #selection="{ item }">
-      <p
-        v-if="(item.raw.name || '').trim() !== ''"
-        class="mb-0"
+  <div>
```
Comment (3asm) [inline project/scanners/components/ScannerSelect.vue:2]: Not the correct UI. You should have a drop down. And in the top of the drop down a button add scan. Click on it and it shows the modal.

**Review by 3asm — DISMISSED** [review]

**Review by benyissa — COMMENTED** [review]
I would simply inform the user that there are no scanners available yet and provide a button that redirects to the scanners page.

**Review by ohachimOs — APPROVED** [review]

**Review by rabsondev — COMMENTED** [review]
Just a small change...**Save** should only be enabled when both the _endpoint_ and _API key_ are provided.
image

**Review by PiranhaSa — APPROVED** [review]

**Review by ybadaoui-ostorlab — APPROVED** [review]

**Review by nmasdoufi-ol — APPROVED** [review]

**Pull request merged**